### ISSUE TITLE
Tutorial - Unable to Run the Harmonize workflow

### ISSUE THREAD

**Issue by nfitbh** [marklogic-community/marklogic-data-hub#643]
#### The issue

Short description of the problem:

Following the tutorial: https://marklogic-community.github.io/marklogic-data-hub/tutorial/harmonizing-product-data/ when I click the "Run Harmonize", the job is created but never succeeds.

Stack Trace available in the log attached.

I've setup Mark Logic, all seem to be working fine; setup DH, again all working fine.  Can import content into staging, browse it, search it, no problems.  The only issue appears to be the harmonize step.  Below is the log (with large stack trace) and the harmonize "CONTENT" javascript.

#### Tech details

Which Operating System are you using?

MACOS 10.13.2.  I'm not using any of the VMs for this, both Mark Logic and DH are running in the MAC OSX environment.

Which version of MarkLogic are you using?

9.0-3.1

Which version of the Data Hub Framework are you using?

quick-start-2.0.2.war

Is this a QuickStart UI bug? If so, which browser are you using?

Pretty sure it's not UI, but in case: Chrome 63.0.3239.84

Also Java version is 9.0.1

#### The devil is in the details...

  `java -jar quickstart.war -d > quickstart.log`

[quickstart.log](https://github.com/marklogic-community/marklogic-data-hub/files/1620902/quickstart.log)

[harmonize.js.txt](https://github.com/marklogic-community/marklogic-data-hub/files/1620909/harmonize.js.txt)

**Comment by nfitbh** [comment]
I should also mention that the following is output to STDERR when starting up DH (not sure if this is normal):

`WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (jar:file:/Users/wilkinsonn7l/dev/ml-guide/ml-data-hub/quick-start-2.0.2.war!/WEB-INF/lib/groovy-2.4.12.jar!/) to method java.lang.Object.finalize()
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
QuickStart UI is Ready and Listening on port 8080.
`

**Comment by nfitbh** [comment]
Stuck job:
![image](https://user-images.githubusercontent.com/3003647/34806410-c8a1c14a-f6d6-11e7-9054-5a46b9614a99.png)

Stuck Job output:
![image](https://user-images.githubusercontent.com/3003647/34806417-da591e42-f6d6-11e7-96f6-f21e932c7c97.png)

Trace:
![image](https://user-images.githubusercontent.com/3003647/34806427-ee991d8a-f6d6-11e7-8e41-4fddd38548ab.png)

**Comment by popzip** [comment]
related to #637?

**Comment by paxtonhare** [comment]
We should probably run a test on Java 9. We've been only testing 8

**Comment by popzip** [comment]
Would this be something @ayuwono could add?

**Comment by nfitbh** [comment]
@popzip I don't think so, #637 is related to "create", mine is upon "run".  I can create the harmonize workflow without any problem, just cannot run it.  Thanks for looking into this.

**Comment by popzip** [comment]
Can you try with Java 8?
